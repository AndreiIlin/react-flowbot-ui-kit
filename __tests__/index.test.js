import '@testing-library/jest-dom';
import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import createFlowBot from '../src/init.jsx';
import config from '../src/examples/config.js';

const initialHtml = '<div id="container"></div>';

beforeEach(async () => {
  document.body.innerHTML = initialHtml;
});

test('originally should be deactivated', async () => {
  const bot = await createFlowBot(config);
  const { container } = render(bot);
  expect(container).toMatchSnapshot();
});

test('toggle bot mode', async () => {
  const user = userEvent.setup();
  const bot = await createFlowBot(config);
  render(bot, { container: document.getElementById('container') });

  const botToggler = screen.getByRole('button');

  await user.click(botToggler);

  expect(document.getElementById('container')).toMatchSnapshot();

  await user.click(botToggler);

  expect(document.getElementById('container')).toMatchSnapshot();
});

test('conversation with the bot', async () => {
  const user = userEvent.setup();
  const bot = await createFlowBot(config);
  render(bot, { container: document.getElementById('container') });

  const botToggler = screen.getByRole('button');
  await user.click(botToggler);

  const questionButton1 = screen.getByRole('button', { name: /Show engine type/i });
  await user.click(questionButton1);

  expect(document.getElementById('container')).toMatchSnapshot();

  await waitFor(() => {
    expect(screen.getByRole('button', { name: /show constraction/i })).toBeInTheDocument();
    expect(document.getElementById('container')).toMatchSnapshot();
  });

  const questionButton2 = screen.getByRole('button', { name: /show constraction/i });
  await user.click(questionButton2);

  await waitFor(() => {
    expect(screen.getByRole('button', { name: /Show engine type/i })).toBeInTheDocument();
    expect(document.getElementById('container')).toMatchSnapshot();
  });
});
