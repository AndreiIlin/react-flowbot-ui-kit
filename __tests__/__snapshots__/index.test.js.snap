// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`conversation with the bot 1`] = `
<div
  id="container"
>
  <div
    class="position-absolute overflow-auto shadow p-3 mb-5 bg-body rounded flowbot-message-box container-sm"
    id="containerElement"
  >
    <div
      class="row"
    >
      <div
        class="col-md-8"
      >
        Hello! I am here to help you!
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-md-8"
      >
        Please, choise a question you want to ask
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="text-end col-md-8 offset-md-4"
      >
        Show engine type
      </div>
    </div>
    <div
      class="d-flex flex-column"
    >
      <div
        class="spinner-border"
        role="status"
      >
        <span
          class="visually-hidden"
        >
          Loading...
        </span>
      </div>
    </div>
  </div>
  <button
    class="position-absolute flowbot-toggler btn btn-primary"
    type="button"
  >
    Active! Click to deactivate
  </button>
</div>
`;

exports[`conversation with the bot 2`] = `
<div
  id="container"
>
  <div
    class="position-absolute overflow-auto shadow p-3 mb-5 bg-body rounded flowbot-message-box container-sm"
    id="containerElement"
  >
    <div
      class="row"
    >
      <div
        class="col-md-8"
      >
        Hello! I am here to help you!
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-md-8"
      >
        Please, choise a question you want to ask
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="text-end col-md-8 offset-md-4"
      >
        Show engine type
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-md-8"
      >
        v8 turbo-diesel
      </div>
    </div>
    <div
      class="d-flex flex-column"
    >
      <button
        class="mb-2 btn btn-primary"
        type="button"
      >
        Show wheels
      </button>
      <button
        class="mb-2 btn btn-primary"
        type="button"
      >
        Show breaks
      </button>
      <button
        class="mb-2 btn btn-primary"
        type="button"
      >
        show constraction
      </button>
    </div>
  </div>
  <button
    class="position-absolute flowbot-toggler btn btn-primary"
    type="button"
  >
    Active! Click to deactivate
  </button>
</div>
`;

exports[`conversation with the bot 3`] = `
<div
  id="container"
>
  <div
    class="position-absolute overflow-auto shadow p-3 mb-5 bg-body rounded flowbot-message-box container-sm"
    id="containerElement"
  >
    <div
      class="row"
    >
      <div
        class="col-md-8"
      >
        Hello! I am here to help you!
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-md-8"
      >
        Please, choise a question you want to ask
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="text-end col-md-8 offset-md-4"
      >
        Show engine type
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-md-8"
      >
        v8 turbo-diesel
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="text-end col-md-8 offset-md-4"
      >
        show constraction
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-md-8"
      >
        Here is car parts whitch you can study
      </div>
    </div>
    <div
      class="d-flex flex-column"
    >
      <button
        class="mb-2 btn btn-primary"
        type="button"
      >
        Show engine type
      </button>
      <button
        class="mb-2 btn btn-primary"
        type="button"
      >
        Show breaks
      </button>
      <button
        class="mb-2 btn btn-primary"
        type="button"
      >
        Show wheels
      </button>
    </div>
  </div>
  <button
    class="position-absolute flowbot-toggler btn btn-primary"
    type="button"
  >
    Active! Click to deactivate
  </button>
</div>
`;

exports[`originally should be deactivated 1`] = `
<div>
  <button
    class="position-absolute flowbot-toggler btn btn-primary"
    type="button"
  >
    Click to activate
  </button>
</div>
`;

exports[`toggle bot mode 1`] = `
<div
  id="container"
>
  <div
    class="position-absolute overflow-auto shadow p-3 mb-5 bg-body rounded flowbot-message-box container-sm"
    id="containerElement"
  >
    <div
      class="row"
    >
      <div
        class="col-md-8"
      >
        Hello! I am here to help you!
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-md-8"
      >
        Please, choise a question you want to ask
      </div>
    </div>
    <div
      class="d-flex flex-column"
    >
      <button
        class="mb-2 btn btn-primary"
        type="button"
      >
        Show engine type
      </button>
      <button
        class="mb-2 btn btn-primary"
        type="button"
      >
        Show wheels
      </button>
      <button
        class="mb-2 btn btn-primary"
        type="button"
      >
        Show breaks
      </button>
    </div>
  </div>
  <button
    class="position-absolute flowbot-toggler btn btn-primary"
    type="button"
  >
    Active! Click to deactivate
  </button>
</div>
`;

exports[`toggle bot mode 2`] = `
<div
  id="container"
>
  <button
    class="position-absolute flowbot-toggler btn btn-primary"
    type="button"
  >
    Click to activate
  </button>
</div>
`;
